### PR TITLE
Fixes Diona nymphs speaking in monkeytalk

### LINKED ISSUE #7678: Fully grown dionae cannot understand diona nymph speak
(WEB REPORT BY: pseudonym REMOTE: 158.69.55.2:7777)

> Revision (Should be above if you're viewing this from ingame!)
> 48128950c9eeee69fccfb8ec45b97c3a7a5ab723
> 
> General description of the issue
> Big Strong Oak chimpers, "*****\* s****"
> 
> What you expected to happen
> Big Strong Oak chimpers, "*****\* s****"
> 
> What actually happened
> Big Strong Oak chimpers, "fuck you slut"
> 
> Steps to reproduce if possible
> Be a diona
> evolve
> talk to a nymph

### PR DESCRIPTION
**Pull request by 9600bauds**
Fixes #7678 

Fun fact: The code for talking in rootspeak is ":q faggot"

### CODE REVIEW

File: code/modules/mob/living/carbon/monkey/diona.dm [hunk 1]
```diff
@@ -163,13 +165,15 @@
 	if(donors.len == 5)
 		ready_evolve = 1
 		to_chat(src, "<span class='good'>You feel ready to move on to your next stage of growth.</span>")
-	else if(donors.len == 2)
-		to_chat(src, "<span class='good'>You feel your awareness expand, and realize you know how to understand the creatures around you.</span>")
 	else if(donors.len == 4)
```
Comment (ghost) [inline code/modules/mob/living/carbon/monkey/diona.dm:168]: Use a switch statement for this whole thing you lazy hallion

**Comment by ghost** [comment]
:+1:

File: code/modules/mob/living/carbon/monkey/diona.dm (hunk 1 above)
Comment (9600bauds) [inline code/modules/mob/living/carbon/monkey/diona.dm:168]: no it was like that when I found it
only riri can tell me to fix it at this point and I won't listen to him anyways

**Comment by Ephx1** [comment]
Fix this one too! #4355

**Comment by 9600bauds** [comment]
How did this feature use to work?

**Comment by Ephx1** [comment]
Sap blood, learn language. Specifically only those known by the person in question.
Although I'd suggest making it so you require multiple blood suckings to get the language, like how it works for Sol Common.

File: code/modules/mob/living/carbon/monkey/diona.dm (hunk 1 above)
Comment (gbasood) [inline code/modules/mob/living/carbon/monkey/diona.dm:168]: fix it nerd

**Comment by gbasood** [comment]
The way it works right now:
- Acquire X samples, learn Sol Common. Hardcoded. You will never learn anything but Sol Common.

It'd be cool if:
- Every sample acquired counted towards learning the languages of the sample donor.

**Comment by 9600bauds** [comment]
2 blood samples per language?

**Comment by ihadtoregisterforthis** [comment]
Two to understand, three to talk is how it works now, but it would be separated for each.
Also, should the language come from the race or the known languages? Taking a sip from a grown Diona would still be useful in that sense.

**Comment by 9600bauds** [comment]
I'll probably do that feature after #7978 gets merged desu

**Comment by 9600bauds** [comment]
wrong button

**Comment by 9600bauds** [comment]
Confrictu fixed

**Comment by gbasood** [comment]
> I'll probably do that feature after #7978 gets merged desu

**WHEN**

**Comment by 9600bauds** [comment]
Who are you quoting

**Comment by gbasood** [comment]
You, dummy

**Comment by 9600bauds** [comment]
It's a meme you dip

**Pull request merged**